packages/store/tests/state.spec.ts

```diff
@@ -140,6 +140,48 @@ describe('State', () => {
       expect(TestBed.get(Store).snapshot().foo).toEqual(['initState', 'onInit']);
     });
 
+    it('should call an UpdateState action handler with multiple states', () => {
```

Comment:
I don't quite understand how this test covers what your  pull request does

I thought you would just add

```ts
it(' ... ', fakeAsync(() => {

  store.dispatch(new UpdateState([ 'MY_PAYLOAD' ]));
  tick(100);

  actions$
      .pipe(ofActionDispatched(UpdateState))
      .subscribe(({ payload }) => expect(payload).toEqual('MY_PAYLOAD'));

}))
```

Reply:
This unit tests shouldn't be dependent on the actions stream.
Take a look at the other tests in the file.

Reply:
You can always create a separate file with the test, but your code has not tested this case.

Reply:
@splincode wouldn't it be the same what he tests right now? I mean maybe I'm unaware of something in this discussion, but it seems like the result will be the same, no?

Reply:
ok